README.md

```diff
@@ -28,26 +28,40 @@ Example for configuring only `pip`:
         uses: blackboard-innersource/gh-action-setup-artifactory@v1
         env:
           ARTIFACTORY_SETUP_NPM: false
+          ARTIFACTORY_YARN_SETUP: false
           ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
           ARTIFACTORY_TOKEN: ${{ secrets.ARTIFACTORY_TOKEN }}
           ARTIFACTORY_PYPI_INDEX: ${{ secrets.ARTIFACTORY_PYPI_INDEX }}
 ```
 
-Example for configuring only `npm`:
+Example for configuring only `npm/yarn1`:
 
 ```yaml
       - name: Setup Artifactory
         uses: blackboard-innersource/gh-action-setup-artifactory@v1
         env:
           ARTIFACTORY_SETUP_PIP: false
+          ARTIFACTORY_YARN_SETUP: false
           ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
           ARTIFACTORY_TOKEN: ${{ secrets.ARTIFACTORY_TOKEN }}
           ARTIFACTORY_NPM_REGISTRY: ${{ secrets.ARTIFACTORY_NPM_REGISTRY }}
 ```
 
-Additional environment variables:
+Example for configuring only `yarn2+`:
+
+```yaml
+      - name: Setup Artifactory
+        uses: blackboard-innersource/gh-action-setup-artifactory@v1
+        env:
+          ARTIFACTORY_SETUP_PIP: false
+          ARTIFACTORY_SETUP_NPM: false
+          ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+          ARTIFACTORY_TOKEN: ${{ secrets.ARTIFACTORY_TOKEN }}
+```
 
-- `ARTIFACTORY_NPM_SCOPES` adds a scope to the NPM credential setup. Set multiple scopes with: `"@scope1,@scope2"`
+Additional environment variables:
+- `ARTIFACTORY_NPM_SCOPES` adds a scope to the NPM/yarn1 credential setup. Set multiple scopes with: `"@scope1,@scope2"`
+  - DO NOT USE for yarn2+ - both repository and scopes should be configured in project's .yarnrc.yml file
```

Comment:
Ah, just saw this line - why wouldn't we configure this for everyone?

Reply:
because if i understand the documentation properly (https://docs.npmjs.com/cli/v8/using-npm/scope) 

it would make sense if we'd use multiple distinct npm registries (different URL) in one repo - imagine if there were separate artifactory instances for fnds and learn. 

it would even make sense if we'd hosted only our packages at artifactory and pulled everything else from https://registry.npmjs.org/, however we use our artifactory as proxy for those too (remote registries), so once again, it's not utilized.

I guess it would make sense if some of our pipelines/ghas would build multiple packages and publish them to different repositories in singular run, to distinguish where should each package be pushed, but that would mean that some company package is published not into artifactory, which would be a red flag ? 

if i'm wrong, I'll be glad if someone who understands this more and has experience with scopes can actually explain

Reply:
The scopes is used when you don't want to proxy everything through Artifactory, but instead only send requests for specific namespaces to Artifactory.

I don't think it is mandatory to get this PR through, but was curious.

I do find it odd that the repository URL isn't set through this, but again, I don't work with yarn and maybe that's not normal.

Reply:
yarn2+ actually combines values from project's and user's .yarnrc.yml files 
we already have repository URL shared in project's [file](https://github.com/blackboard-foundations/bbdata-ui/pull/210/files#diff-88fbe28c4102501b94961511a0d70ff895bf39970b4d3fc11917794a239117c5) (and some other things), so we only need creds in the user's file (which is generated by this project for CI and manually for local devices)
runtime uses values from both locations combined ... i guess if there was value for the same key in both, one will have precedence 

IMHO using artifactory for everything is a bit safer, if some package(version) gets pulled down from https://registry.npmjs.org/, we still can get to it as long as it's cached on artifactory.

not sure about the network performance difference, but i guess safety overules speed in this case